### PR TITLE
[7.0] Replacing current decorators

### LINKED ISSUE #5283: [7.0] move legacy-decorators into the repo, replacing current decorators
https://github.com/loganfsmyth/babel-plugin-transform-decorators-legacy

Basically since the [current decorators transform](https://github.com/babel/babel/blob/f8ffe03e79d95b2b9ed3481dbb3e90b76c536d90/packages/babel-plugin-transform-decorators/src/index.js#L74-L86) throws an error, we should just move the implementation (even though it's the older stage-1 spec) into the babel repo. We'll do this in Babel 7 so it doesn't cause issues for anyone

We'll also want to change stage-1 to refer to the current transform again (reverting the previous pr) but we can do that.

---

- [ ] Comment below you are going to do this (for others to know)
- [ ] Read [CONTRIBUTING.md](https://github.com/babel/babel/blob/master/CONTRIBUTING.md)
- [ ] remember to run `make build` after a source code change or better just run `make watch` in another terminal window to auto run babel while you are developing
- [ ] Check you are on 7.0 branch with `git checkout 7.0` and then make a branch off of that to PR
- [ ] Copy the contents of https://github.com/loganfsmyth/babel-plugin-transform-decorators-legacy into the transform-decorators folder under the `packages/` directory. Try to keep the structure of the current repo but then replacing with src/tests/readme and merge the correct things where possible.
- [ ] might need to change the structure of the tests to match how babel does things via fixtures, etc but we can help with this

### PR DESCRIPTION
**Pull request by alxpy**
| Q                        | A 
| ------------------------ | ---
| Patch: Bug Fix?          | no
| Major: Breaking Change?  | no
| Minor: New Feature?      | yes
| Deprecations?            | no
| Spec Compliancy?         | yes
| Tests Added/Pass?        | yes
| Fixed Tickets            | `Close #5283` 
| License                  | MIT
| Doc PR                   | 
| Dependency Changes       | 

I started working on https://github.com/babel/babel/issues/5283 and I have some doubts about the tests. Please look at the tests structure.  If all is well, I move other tests.

### CODE REVIEW

**Comment by alxpy** [comment]
@hzoo So? Did I make everything right? Can I continue?

**Comment by xtuc** [comment]
I updated the README to be consistent with the other packages.

File: packages/babel-plugin-transform-decorators/README.md [hunk 1]
```diff
@@ -46,24 +46,42 @@ function enumerable(value) {
 }
 ```
 
-## Installation
+## Installation & Usage
```
Comment (existentialism) [inline packages/babel-plugin-transform-decorators/README.md]: Nit: to keep it consistent with the rest of the docs let's keep Installation and Usage as separate sections

File: packages/babel-plugin-transform-decorators/README.md [hunk 2]
```diff
@@ -46,24 +46,42 @@ function enumerable(value) {
...
 
 ```sh
-npm install --save-dev babel-plugin-transform-decorators
+npm install --save-dev babel-plugin-transform-decorators`
```
Comment (xtuc) [inline packages/babel-plugin-transform-decorators/README.md]: Oops, I added a trailing backtick there.

**Comment by existentialism** [comment]
Nice work @alxpy!

For the tests, it may be better if we break each test up into its own folder/exec so any specific failure would be easy to identify.

```
|- fixtures
|     |- class
|           |- descriptor-expr-order
|           |     |- exec.js
|           |- method-reverse-order
|           |     |- exec.js
```

**Comment by alxpy** [comment]
@existentialism I have tried to do like you wrote, but with this structure, `expected.js` was created =(

**Comment by alxpy** [comment]
@existentialism sorry, I was wrong. I forgot to remove the `exec` folder

**Comment by alxpy** [comment]
@existentialism so, I think I moved all tests

File: packages/babel-plugin-transform-decorators/README.md [hunk 3]
```diff
@@ -54,16 +54,36 @@ npm install --save-dev babel-plugin-transform-decorators
...
 }
 ```
 
+#### NOTE: Order of Plugins Matters!
+
+If you are including your plugins manually and using `transform-class-properties`, make sure that `transform-decorators` comes *before* `transform-class-properties`.
+
+Wrong:
+
+```js
```
Comment (hzoo) [inline packages/babel-plugin-transform-decorators/README.md]: Can we either not make this `js` or change it to json with wrapping `{}`?

Comment (alxpy) [inline packages/babel-plugin-transform-decorators/README.md]: yes

**Comment by alxpy** [comment]
I finished work. Do I need to fix something?

File: packages/babel-plugin-transform-decorators/package.json [hunk 4]
```diff
@@ -1,17 +1,20 @@
...
+    "babel",
+    "babel-plugin",
+    "es7",
+    "decorators"
   ],
   "dependencies": {
-    "babel-types": "^6.22.0",
     "babel-plugin-syntax-decorators": "^6.13.0",
-    "babel-helper-explode-class": "^6.22.0",
+    "babel-runtime": "^6.2.0",
```
Comment (hzoo) [inline packages/babel-plugin-transform-decorators/package.json]: note: after this is merged, we should remove this dep in 7.0

**Comment by hzoo** [comment]
Nope just a few docs things I'l do when merging, thanks!

**Comment by hzoo** [comment]
Really nice work @alxpy! looking forward to more contributors 😄

**Comment by alxpy** [comment]
@hzoo I was glad to help! I think that I will be helping again.

**Pull request merged**